pkgs/development/python-modules/pyotb/default.nix

```
@@ -0,0 +1,98 @@
+{
+  lib,
+  pkgs,
+  fetchFromGitHub,
+  buildPythonPackage,
+
+  # build-system
+  setuptools,
+
+  # dependencies
+  otb,
+
+  # tests
+  pytestCheckHook,
+  pytest-cov,
+  requests,
+  writableTmpDirAsHomeHook,
+}:
+let
+  # fetch the test data separately or else none of the test will work
+  # https://github.com/orfeotoolbox/pyotb/blob/develop/tests/tests_data.py
+  spotImage = pkgs.fetchurl {
+    url = "https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/raw/develop/Data/Input/SP67_FR_subset_1.tif";
+    sha256 = "sha256-MuWY/g7KI+F23lFY/+AX5MLWJlIgHCr5BvFjDHzpWgY=";
+  };
+
+  pleiadesImage = pkgs.fetchurl {
+    url = "https://gitlab.orfeo-toolbox.org/orfeotoolbox/otb/-/raw/develop/Data/Baseline/OTB/Images/prTvOrthoRectification_pleiades-1_noDEM.tif";
+    sha256 = "sha256-1EsGAJdHgBIb/gfbh4Y7yEEmYHb54bSx4fEMKssZ/oA=";
+  };
+in
+buildPythonPackage rec {
+  pname = "pyotb";
+  version = "2.1.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "orfeotoolbox";
+    repo = "pyotb";
+    tag = version;
+    hash = "sha256-KomIMVx4jfsTSbGtoml9ON/82sHanOkp/mp1TiUaa2E=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  postPatch = ''
+    # Replace OTB_ROOT with the actual path from nix build (so one doesn't have to set it during runtime)
+    substituteInPlace pyotb/helpers.py --replace-fail 'OTB_ROOT = os.environ.get("OTB_ROOT")' 'OTB_ROOT = "${otb}"'
+
+    # Remove setting these path as we have build OTB from source and these are statically linked, no need to set these environment
```

```suggestion
  postPatch = 
    # Replace OTB_ROOT with the actual path from nix build (so one doesn't have to set it during runtime)
    ''  
      substituteInPlace pyotb/helpers.py --replace-fail 'OTB_ROOT = os.environ.get("OTB_ROOT")' 'OTB_ROOT = "${otb}"'
    ''
    # Remove setting these path as we have build OTB from source and these are statically linked, no need to set these environment
    + ''
```
Try to keep comments out of the nix strings.

---

Thanks @GaetanLepage !

I have updated it.